### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Ghiftee**
Here, I implemented the following:

- Create a class Corrector with a method correct_name that does the following:
. Make sure the first letter of the given name is a capital letter.
. Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.
- Set an instance of Corrector in Person on initialization.
- In Person add a method validate_name that uses the Corrector instance to validate and save the corrected name.

Kindly review my code. Thank you!

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Corrector
+  def correct_name(name)
+    capital_name = name.capitalize
+    capital_name.slice(0, 10)
+  end
+end
```
Comment (Emmyn5600) [inline corrector.rb:6]: - [x] Good job here.

ruby-3

File: person.rb [hunk 2]
```diff
@@ -13,6 +15,11 @@ def can_use_services?
     true if is_of_age? || @parent_permission
```
Comment (Emmyn5600) [inline person.rb]: - [x] Here, you are calling `is_of_age?` however, your method is named as `of_age?` it is raising an error
ruby-4

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

### Hi, @Ghiftee 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- You have descriptive Readme file:heavy_check_mark:

   ###  Changes required :recycle: 
   
     - _Check the comment under the review_
    
    Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 2 above)
Comment (Ghiftee) [reply person.rb]: Fixed

File: corrector.rb (hunk 1 above)
Comment (Ghiftee) [reply corrector.rb:6]: Thank you

File: person.rb (hunk 2 above)
Comment (Emmyn5600) [reply person.rb]: Good Job

**Review by Emmyn5600 — APPROVED** [review]
Hi,

Good job so far! 👏🏼 @Ghiftee 


## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewer.

 #### Optional suggestions
 
 - N/A.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**